File: Project Euler/Problem 1 - Multiples of 3 or 5/Answers/Answer.kt
import kotlin.math.pow

fun main() {
    print("Enter a number: ")
    val value: Int = readLine()!!.toInt()
    var res: Int = multiples_of_3_and_5(value)
    println(res)
}

fun multiples_of_3_and_5(limit: Int): Int {
    var new_limit: Int = limit-1
    var (limit3: Double, limit5: Double, limit15: Double) = limits_for_3_5_and_15(new_limit)
    var (summation3: Int, summation5: Int, summation15: Int) = summations_for_3_5_and_15(limit3, limit5, limit15)
    var answer: Int = summation3 + summation5 - summation15
    return (answer)
}

fun limits_for_3_5_and_15(limit: Int): Triple<Double, Double, Double> {
    // Make these doubles as the power operator only works with doubles or floats
    var limit3: Double = limit / 3.0
    var limit5: Double = limit / 5.0
    var limit15: Double  = limit / 15.0
    return Triple(limit3, limit5, limit15)
}

fun summations_for_3_5_and_15(limit3: Double, limit5: Double, limit15: Double): Triple<Int, Int, Int> {
    var summation3: Int = (limit3.pow(2) + limit3).toInt()*3/2
    var summation5: Int = (limit5.pow(2) + limit5).toInt()*5/2
    var summation15: Int = (limit15.pow(2) + limit15).toInt()*15/2
    return Triple(summation3, summation5, summation15)
}

File: Project Euler/Problem 1 - Multiples of 3 or 5/Answers/Answer.swift
import Foundation

func multiples_of_3_and_5(limit: Int) -> Int {
    let new_limit = limit-1
    let (limit3, limit5, limit15)  = limits_for_3_5_and_15(limit: new_limit)
    let (summation3, summation5, summation15) = summations_for_3_5_and_15(limit3: limit3, limit5: limit5, limit15: limit15)
    let answer = summation3 + summation5 - summation15
    return Int(answer)
}

func limits_for_3_5_and_15(limit: Int) -> (Double, Double, Double) {
    let limit3: Double = Double(limit/3)
    let limit5: Double = Double(limit/5)
    let limit15: Double = Double(limit/15)
    return (limit3, limit5, limit15)
}

func summations_for_3_5_and_15(limit3: Double, limit5: Double, limit15: Double) -> (Double, Double, Double) {
    let summation3 = (pow(limit3, 2) + limit3)*3/2
    let summation5 = (pow(limit5, 2) + limit5)*5/2
    let summation15 = (pow(limit15, 2) + limit15)*15/2
    return (summation3, summation5, summation15)
}

print("Enter a number: ", terminator: "")
let value = Int(readLine()!)
print(multiples_of_3_and_5(limit: value!))